Docker.md

```diff
@@ -0,0 +1,126 @@
+| directory | description | notes |
+|---|---|---|
+| /opt/frank/resources | For application-wide properties, may contain files or a .jar with all files | Minimum required properties to set are `instance.name` and `configurations.names`, can also be set using environment variables |
+| /opt/frank/configurations | For configurations, may contain a directory with files per configuration or a .jar containing a directory per configuration | When Configuration.xml is not located at `<configurationName>/Configuration.xml`, your resources should include a property `configurations.<configurationName>.configurationFile` containing the path to the Configuration.xml |
+| /opt/frank/testtool | For Larva tests that are included in the image | |
+| /opt/frank/testtool-ext | For Larva tests that are mounted from the environment | |
+| /usr/local/tomcat/lib | Contains drivers and other dependencies | Contains all Framework required dependencies and drivers for supported JMS and JDBC systems |
+| /usr/local/tomcat/logs | Log directory | |
+
+## Files
+
+The image also contains the following files:
+| file | description | notes |
+|---|---|---|
+| /usr/local/tomcat/conf/Catalina/localhost/ROOT.xml | mount/copy of your context.xml | Use hostname `host.docker.internal` to get to the host machine for local testing. Changing this file will require a new instance to be started, it cannot be reloaded |
+| /usr/local/tomcat/conf/server.xml | mount/copy of your server.xml | Contains the default server.xml of Tomcat, replace to secure your application |
+| /usr/local/tomcat/conf/catalina.properties | Server properties, contains default framework values | Do not replace this file, use [Environment variables](#Environment-variables) or append to the file, see [Dockerfile](docker/appserver/Tomcat/Dockerfile) for an example |
+
+## Permissions
+
+As this image does not run Tomcat using the root user, file permissions need to be set correctly. By default this is done during startup, to ensure that all files copied to the above locations have the correct permissions. For images with a large number of files in these locations, this can reduce startup performance. It is possible to set the permissions during build and disable the step during startup by adding the following lines at the end of your Dockerfile:
+```
+RUN /setPermissions.sh
+ENV SET_PERMISSIONS_ON_STARTUP=FALSE
+```
+
+Logging
+=======
+
+Generated log files are stored in `/usr/local/tomcat/logs`.
+
+Environment variables
+=====================
+
+Environment variables can be used to set parameters. Environment variables have the highest precedence and override parameters set in .property files supplied by Tomcat, resources and configurations.
+
+As `org.apache.tomcat.util.digester.PROPERTY_SOURCE=org.apache.tomcat.util.digester.EnvironmentPropertySource` is set in our images, environment variables can also be used to replace parameters in Tomcat configuration files such as server.xml and context.xml.
```

Comment:
Het gaat natuurlijk met name on het gebruik van de CredentialProvider om je wachtwoorden niet te exposen. Deze properties moeten in credentials.properties staan. Indirect inherit deze file inderdaad van de environment variables.
let wel op, het framework weet niets van de credentials.properties af, je kan er dus alleen properties in zetten voor de CredentialProvider.

Reply:
Om CredentialProvider te kunnen gebruiken moet nl.nn.credentialprovider.CredentialProvidingPropertySource gebruikt worden.